packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol

```diff
@@ -595,3 +595,81 @@ contract L1CrossDomainMessenger_Test is Bridge_Initializer {
         assertEq(l1CrossDomainMessenger.paused(), superchainConfig.paused());
     }
 }
+
+/// @dev A regression test against a reentrancy vulnerability in the CrossDomainMessenger contract, which
+///      was possible by intercepting and sandwhiching a signed Safe Transaction to upgrade it.
+contract L1CrossDomainMessenger_ReinitReentryTest is Bridge_Initializer {
+    bool attacked;
+
+    // Common values used across functions
+    address target;
+    address sender = sender = Predeploys.L2_CROSS_DOMAIN_MESSENGER;
+    uint256 messageValue = 50;
+
+    function setUp() public override {
+        super.setUp();
+        target = address(this);
+        vm.deal(address(l1CrossDomainMessenger), messageValue * 2);
+    }
+
+    /// @dev This method will be called by the relayed message, and will attempt to reenter the relayMessage function
+    ///      exactly once.
+    function reinitAndReenter() public payable {
+        // only attempt the attack once
+        if (!attacked) {
+            // set initialized to false
+            vm.store(address(l1CrossDomainMessenger), 0, bytes32(uint256(0)));
+
+            // call the initializer function
+            l1CrossDomainMessenger.initialize(SuperchainConfig(superchainConfig));
+
+            // attempt to re-replay the withdrawal
```

Comment:
Could we add an expect emit here to clarify what codepath this is supposed to take?

Reply:
resolved here: https://github.com/ethereum-optimism/optimism/pull/8864/files#r1443308813